zerver/models.py

```diff
@@ -1570,6 +1571,7 @@ class UserBaseSettings(models.Model):
         twenty_four_hour_time=bool,
     )
 
+    # Do not add new user settings to legacy dicts!
     notification_settings_legacy = dict(
         desktop_icon_count_display=int,
```

Comment:
For a "don't add" warning comment like this, my suggestion would be to put it right inside the list itself, for maximal proximity to where the change it's warning against would go. I think that maximizes the chance that a contributor seeing the list and contemplating adding to it will see the warning.

I'd also suggest (a) being a little more concrete that the message is talking about *this* list, right here (not something elsewhere that might be called a "legacy dict"), and applies to anything new you might add to it (i.e., that "user settings" isn't a limitation on the scope of the warning); and (b) explicitly pointing at what to do instead.

```suggestion
    notification_settings_legacy = dict(
        # Don't add anything new to this legacy dict.
        # Instead, see `modern_notification_settings` below.
        desktop_icon_count_display=int,
```

Reply:
Yes, that makes a lot of sense. Thanks for the review! I've updated the comments (location and text) and agree that it's much clearer now.